x-pack/plugins/security_solution/common/api/detection_engine/model/rule_schema/common_attributes.schema.yaml

```diff
@@ -6,13 +6,19 @@ paths: {}
 components:
   x-codegen-enabled: true
   schemas:
+    NonEmptyString:
+      type: string
+      pattern: ^(?! *$).+$
```

Comment:
Just curious why using this relatively complex regex while it could be `^[^\s]+$` (require one ore more non space characters). While it's stricter than `NonEmptyString` defined previously in via `io-ts` but we still don't expect any kind of space characters as a rule name.

Reply:
In my honest opinion, we need neither of those regexes 🙂 I'd define a non-empty string as a string with length > 0, regardless of its content.